[Nx Vite TsPaths] Unable to resolve any file

### Current Behavior

When setting up a new React Monorepo with Vite `npx create-nx-workspace@latest react-monorepo --preset=react-monorepo` and then trying to build the App, the logs are cluttered with messages about Vite being unable to resolve paths. 

Example:

[Nx Vite TsPaths] Unable to resolve C:\Users\Me\Data\Code\react-monorepo\apps\my-react-app\index.html with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved C:\Users\Me\Data\Code\react-monorepo\apps\my-react-app\index.html to undefined

### Expected Behavior

All Paths should be resolved correctly.

### GitHub Repo

_No response_

### Steps to Reproduce

1. `npx create-nx-workspace@latest react-monorepo --preset=react-monorepo`
2. make selections: `√ Application name · my-react-app
√ Which bundler would you like to use? · vite
√ Test runner to use for end to end (E2E) tests · cypress
√ Default stylesheet format · tailwind
√ Which CI provider would you like to use? · azure`
4. `cd react-monorepo`
5. `npx nx build my-react-app --verbose`

### Nx Report

```shell
Node           : 22.9.0
OS             : win32-x64
Native Target  : x86_64-windows
npm            : 10.8.3

nx                     : 20.2.2
@nx/js                 : 20.2.2
@nx/jest               : 20.2.2
@nx/eslint             : 20.2.2
@nx/workspace          : 20.2.2
@nx/cypress            : 20.2.2
@nx/devkit             : 20.2.2
@nx/eslint-plugin      : 20.2.2
@nx/module-federation  : 20.2.2
@nx/react              : 20.2.2
@nx/vite               : 20.2.2
@nx/web                : 20.2.2
typescript             : 5.6.3
---------------------------------------
Registered Plugins:
@nx/vite/plugin
@nx/eslint/plugin
@nx/cypress/plugin
@nx/jest/plugin
```

### Failure Logs

```shell
~/Data/Code
$ npx create-nx-workspace@latest react-monorepo --preset=react-monorepo
npm warn exec The following package was not found and will be installed: create-nx-workspace@20.2.2

 NX   Let's create a new workspace [https://nx.dev/getting-started/intro]

√ Application name · my-react-app
√ Which bundler would you like to use? · vite
√ Test runner to use for end to end (E2E) tests · cypress
√ Default stylesheet format · tailwind
√ Which CI provider would you like to use? · azure

 NX   Creating your v20.2.2 workspace.

- Installing dependencies with npm
√ Installing dependencies with npm
- Creating your workspace in react-monorepo
√ Successfully created the workspace: react-monorepo.
- Checking Nx Cloud setup
√ Nx Cloud has been set up successfully
- Generating CI workflow
√ CI workflow has been generated successfully

 NX   Your CI setup is almost complete.

Finish it by visiting: https://cloud.nx.app/connect/...

 NX   Welcome to the Nx community! 👋

🌟 Star Nx on GitHub: https://github.com/nrwl/nx
📢 Stay up to date on X: https://x.com/nxdevtools
💬 Discuss Nx on Discord: https://go.nx.dev/community

~/Data/Code
$ cd react-monorepo

~/Data/Code/react-monorepo (main)
$ npx nx build my-react-app --verbose

> nx run my-react-app:build

> vite build

The CJS build of Vite's Node API is deprecated. See https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

[Nx Vite TsPaths] Found tsconfig at C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] first parsed tsconfig:  {
  resultType: 'success',
  configFileAbsolutePath: 'C:\\Users\\Me\\Data\\Code\\react-monorepo\\tsconfig.base.json',
  baseUrl: '.',
  absoluteBaseUrl: 'C:\\Users\\Me\\Data\\Code\\react-monorepo',
  paths: {},
  addMatchAll: true
}

[Nx Vite TsPaths] Found tsconfig at C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] fallback parsed tsconfig:  {
  resultType: 'success',
  configFileAbsolutePath: 'C:\\Users\\Me\\Data\\Code\\react-monorepo\\tsconfig.base.json',
  baseUrl: '.',
  absoluteBaseUrl: 'C:\\Users\\Me\\Data\\Code\\react-monorepo',
  paths: {},
  addMatchAll: true
}
vite v5.4.11 building for production...

[Nx Vite TsPaths] Unable to resolve C:\Users\Me\Data\Code\react-monorepo\apps\my-react-app\index.html with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved C:\Users\Me\Data\Code\react-monorepo\apps\my-react-app\index.html to undefined

[Nx Vite TsPaths] Unable to resolve /src/styles.css with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved /src/styles.css to undefined
transforming...

[Nx Vite TsPaths] Unable to resolve vite/modulepreload-polyfill with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved vite/modulepreload-polyfill to undefined

[Nx Vite TsPaths] Unable to resolve /src/styles.css with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved /src/styles.css to undefined

[Nx Vite TsPaths] Unable to resolve /src/main.tsx with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved /src/main.tsx to undefined

[Nx Vite TsPaths] Unable to resolve react/jsx-runtime with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved react/jsx-runtime to undefined

[Nx Vite TsPaths] Unable to resolve react with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved react to undefined

[Nx Vite TsPaths] Unable to resolve react-dom/client with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved react-dom/client to undefined

[Nx Vite TsPaths] Unable to resolve ./app/app with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved ./app/app to undefined

[Nx Vite TsPaths] Unable to resolve ./cjs/react-jsx-runtime.production.min.js with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved ./cjs/react-jsx-runtime.production.min.js to undefined

[Nx Vite TsPaths] Unable to resolve react/jsx-runtime with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved react/jsx-runtime to undefined

[Nx Vite TsPaths] Unable to resolve ./nx-welcome with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved ./nx-welcome to undefined

[Nx Vite TsPaths] Unable to resolve ./cjs/react.production.min.js with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved ./cjs/react.production.min.js to undefined

[Nx Vite TsPaths] Unable to resolve react-dom with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved react-dom to undefined

[Nx Vite TsPaths] Unable to resolve react with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved react to undefined

[Nx Vite TsPaths] Unable to resolve react/jsx-runtime with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved react/jsx-runtime to undefined

[Nx Vite TsPaths] Unable to resolve ./cjs/react-dom.production.min.js with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved ./cjs/react-dom.production.min.js to undefined

[Nx Vite TsPaths] Unable to resolve react with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved react to undefined

[Nx Vite TsPaths] Unable to resolve scheduler with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved scheduler to undefined

[Nx Vite TsPaths] Unable to resolve ./cjs/scheduler.production.min.js with tsconfig paths. Using fallback file matching.

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Trying to resolve file from config in C:\Users\Me\Data\Code\react-monorepo\tsconfig.base.json

[Nx Vite TsPaths] Resolved ./cjs/scheduler.production.min.js to undefined
✓ 32 modules transformed.
rendering chunks...
computing gzip size...
../../dist/apps/my-react-app/index.html                   0.48 kB │ gzip:  0.31 kB
../../dist/apps/my-react-app/assets/index-Dob4_79B.css    5.94 kB │ gzip:  1.73 kB
../../dist/apps/my-react-app/assets/index-CwAXzi5-.js   166.81 kB │ gzip: 51.85 kB
✓ built in 4.81s



 NX   Successfully ran target build for project my-react-app


View logs and investigate cache misses at https://nx.app/runs/wtLwu3F3do
```

### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [x] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

I've had this error since bunping to NX v20.x.x. But it has not bothered me since I silenced the waring by removing --verbose. But today, migrating to nx v20.3.1 the warning became an error and stopped my Storybook build saying error TS5083: Cannot read file './tsconfig.json'. I'm currently running Storybok v8.4.5 and vite v5.4.8.

```
> nx run design-system-storybook-react:build
 NX  Storybook builder starting ...
info => Cleaning outputDir: dist\apps\design-system-storybook-react
The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.
info => Loading presets
info => Building manager..
info
 => Manager built (235 ms)
info => Building preview..
info => Copying static files: apps\design-system-storybook-react\assets at dist\apps\design-system-storybook-react
[Nx Vite TsPaths] Found tsconfig at C:\Projects\Designsystem\DS-Mono-branches\tsconfig.base.json
[Nx Vite TsPaths] first parsed tsconfig:  {
  resultType: 'success',
  configFileAbsolutePath: 'C:\\Projects\\Designsystem\\DS-Mono-branches\\tsconfig.base.json',
  baseUrl: '.',
  absoluteBaseUrl: 'C:\\Projects\\Designsystem\\DS-Mono-branches',
  paths: {
    '@namespace/design-system-icons': [ 'libs/design-system-icons/index.ts' ],
    '@namespace/design-system-storybook-react': [ 'apps/design-system-storybook-react/src/index.ts' ],
    '@namespace/design-system-tokens': [ 'libs/design-system-tokens/dist/ts/tokens.ts' ],
    '@namespace/design-system-ui-react': [ 'libs/design-system-ui-react/src/index.ts' ],
    '@namespace/design-system-ui-web': [ 'libs/design-system-ui-web/src/index.ts' ],
    '@namespace/design-system-utils': [ 'libs/design-system-utils/src/index.ts' ]
  },
  addMatchAll: true
}
[Nx Vite TsPaths] Found tsconfig at C:\Projects\Designsystem\DS-Mono-branches\tsconfig.base.json
[Nx Vite TsPaths] fallback parsed tsconfig:  {
  resultType: 'success',
  configFileAbsolutePath: 'C:\\Projects\\Designsystem\\DS-Mono-branches\\tsconfig.base.json',
  baseUrl: '.',
  absoluteBaseUrl: 'C:\\Projects\\Designsystem\\DS-Mono-branches',
  paths: {
    '@namespace/design-system-icons': [ 'libs/design-system-icons/index.ts' ],
    '@namespace/design-system-storybook-react': [ 'apps/design-system-storybook-react/src/index.ts' ],
    '@namespace/design-system-tokens': [ 'libs/design-system-tokens/dist/ts/tokens.ts' ],
    '@namespace/design-system-ui-react': [ 'libs/design-system-ui-react/src/index.ts' ],
    '@namespace/design-system-ui-web': [ 'libs/design-system-ui-web/src/index.ts' ],
    '@namespace/design-system-utils': [ 'libs/design-system-utils/src/index.ts' ]
  },
  addMatchAll: true
}
error TS5083: Cannot read file './tsconfig.json'.
 NX   Running target build for project design-system-storybook-react and 6 tasks it depends on failed
Failed tasks:
- design-system-storybook-react:build
Hint: run the command with --verbose for more details.
```

--- a/packages/vite/plugins/nx-tsconfig-paths.plugin.ts
+++ b/packages/vite/plugins/nx-tsconfig-paths.plugin.ts
@@ -24,8 +24,8 @@ import { isUsingTsSolutionSetup } from '@nx/js/src/utils/typescript/ts-solution-
 export interface nxViteTsPathsOptions {
   /**
    * Enable debug logging
-   * If set to 'force-ignore', it will always ignore the debug logging even when `--verbose` or `NX_VERBOSE_LOGGING` is set to true.
-   * @default false
+   * If set to false, it will always ignore the debug logging even when `--verbose` or `NX_VERBOSE_LOGGING` is set to true.
+   * @default undefined
    **/
   debug?: boolean;
   /**

--- a/packages/vite/plugins/nx-tsconfig-paths.plugin.ts
+++ b/packages/vite/plugins/nx-tsconfig-paths.plugin.ts
@@ -224,11 +224,7 @@ export function nxViteTsPaths(options: nxViteTsPathsOptions = {}) {
   }
 
   function logIt(...msg: any[]) {
-    const shouldLog =
-      options?.debug === false
-        ? false
-        : process.env.NX_VERBOSE_LOGGING === 'true' || options?.debug;
-    if (shouldLog) {
+    if (process.env.NX_VERBOSE_LOGGING === 'true' && options?.debug !== false) {
       console.debug('\n[Nx Vite TsPaths]', ...msg);
     }
   }

--- a/packages/vite/plugins/nx-tsconfig-paths.plugin.ts
+++ b/packages/vite/plugins/nx-tsconfig-paths.plugin.ts
@@ -24,6 +24,7 @@ import { isUsingTsSolutionSetup } from '@nx/js/src/utils/typescript/ts-solution-
 export interface nxViteTsPathsOptions {
   /**
    * Enable debug logging
+   * If set to 'force-ignore', it will always ignore the debug logging even when `--verbose` or `NX_VERBOSE_LOGGING` is set to true.
    * @default false
    **/
   debug?: boolean;
@@ -223,7 +224,11 @@ export function nxViteTsPaths(options: nxViteTsPathsOptions = {}) {
   }
 
   function logIt(...msg: any[]) {
-    if (process.env.NX_VERBOSE_LOGGING === 'true' || options?.debug) {
+    const shouldLog =
+      options?.debug === false
+        ? false
+        : process.env.NX_VERBOSE_LOGGING === 'true' || options?.debug;
+    if (shouldLog) {
       console.debug('\n[Nx Vite TsPaths]', ...msg);
     }
   }

